aten/src/ATen/native/cuda/Sort.cu

```diff
@@ -85,6 +267,31 @@ std::tuple<Tensor &,Tensor &> sort_out_stable_cuda(const Tensor & self, c10::opt
 
   TORCH_CHECK(nsort <= std::numeric_limits<int>::max(),
     "The dimension being sorted can not have more than INT_MAX elsments.");
+  // use inplace algorithm for smaller input sizes without stable=True
+  if (should_use_small_sort(self, dim) && !stable.value()) {
+    // from thc: sorted->values, indices->indices, input->self
+
+    if (!values.defined()) {
+      values = at::empty_like(self);
+    }
+    if (!indices.defined()) {
+      indices = at::empty_like(self, self.options().dtype(kLong));
+    }
+
+    // Make sure sufficient output space is allocated
+    std::vector<int64_t> self_size = self.sizes().vec();
+    at::native::resize_output(values, self_size);
+    at::native::resize_output(indices, self_size);
+    fillSliceWithIndex(indices, dim);
+
+    // We sort k/v pairs in-place; copy unsorted input to output
+    values.copy_(self);
+
+    // Sort using our in-place k/v kernel that supports arbitrary
+    // layout
+    sortKeyValueInplace(values, indices, dim, descending);
+    return std::forward_as_tuple(values, indices);
+  }
 
   if (ndim == 0) {
```

Comment:
This part of code handles 0D tensor, so you don't need to handle it in `sortKeyValueInplace`